types/node-wav/node-wav-tests.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright 2016 Andreas Gal
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+// The following has been altered from the
+// [original version](https://github.com/andreasgal/node-wav/commit/9804a3fcbb251adb2766a5dad739b1ed033f3ba5?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5)
+// to fit within DefinitelyTyped lint requirements.
+
```

Comment:
The test is based on the example from the node-wave repository which is MIT licensed, the license including:

> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

I would therefore expect to need to include the license alongside the test.

Reply:
you can use this:
```ts
import wav = require('node-wav');

declare const buffer: Buffer;

// $ExpectType { readonly sampleRate: number; readonly channelData: readonly Float32Array[]; }
const wavFileInfo = wav.decode(buffer);
// $ExpectType Buffer
wav.encode(wavFileInfo.channelData, {
    sampleRate: wavFileInfo.sampleRate,
    float: true,
    bitDepth: 32,
    floatinPoint: true,
});
// $ExpectType Buffer
wav.encode(wavFileInfo.channelData, {});
// $ExpectError Expected 2 arguments, but got 1.
wav.encode(wavFileInfo.channelData);
```

Note that you're missing `floatingPoint` option for encode:
https://github.com/andreasgal/node-wav/blob/master/index.js#L237

Reply:
Thank you, I've added the `floatingPoint` option and replaced the test with yours.  I had to drop `Expected 2 arguments, but got 1.` from the final `$ExpectError` to get the tests to pass, but it otherwise looked good, thanks!